src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions_Negate.cs

```diff
@@ -412,21 +442,46 @@ private static IOperation RemoveImplicitConversion(IOperation operation)
 
         private static SyntaxNode GetNegationOfUnaryPattern(
             SyntaxNode pattern,
+            SyntaxGenerator generator,
             SyntaxGeneratorInternal generatorInternal,
             ISyntaxFacts syntaxFacts)
         {
-            syntaxFacts.GetPartsOfUnaryPattern(pattern, out var opToken, out var subPattern);
-
             // not not p    ->   p
             if (syntaxFacts.IsNotPattern(pattern))
             {
-                return subPattern.WithPrependedLeadingTrivia(opToken.LeadingTrivia)
-                                 .WithAdditionalAnnotations(Simplifier.Annotation);
+                return GetNegationOfNotPattern(pattern, generator, generatorInternal, syntaxFacts);
             }
 
             // If there are other interesting unary patterns in the future, we can support specialized logic for
             // negating them here.
             return generatorInternal.NotPattern(pattern);
         }
+
+        private static SyntaxNode GetNegationOfNotPattern(
+            SyntaxNode pattern,
+            SyntaxGenerator generator,
+            SyntaxGeneratorInternal generatorInternal,
+            ISyntaxFacts syntaxFacts)
+        {
+            Contract.ThrowIfFalse(syntaxFacts.IsNotPattern(pattern));
+
+            syntaxFacts.GetPartsOfUnaryPattern(pattern, out var opToken, out var subPattern);
+
+            // If we started with `not object`, instead of converting to `object`, directly convert to `not null`
```

Comment:
I can't decide how I feel about this. `is not object` has been valid as long as `is null`, right? So someone writing it presumably did so deliberately, so isn't the negation of it `is object`?

I guess we'll hear about it if anyone doesn't like it :)

Reply:
> I can't decide how I feel about this. `is not object` has been valid as long as `is null`, right?

Nope!  Patterns were added in C#7.   `not` patterns though only came in C#9.